Tests/Unit/NetworkServer/BasicsStation/BasicsStationNetworkServerTests.cs

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#nullable enable
+
+namespace LoRaWan.Tests.Unit.NetworkServer.BasicsStation
+{
+    using System;
+    using System.IO;
+    using System.Security.Cryptography;
+    using System.Security.Cryptography.X509Certificates;
+    using System.Threading.Tasks;
+    using LoRaWan.NetworkServer;
+    using LoRaWan.NetworkServer.BasicsStation;
+    using Microsoft.AspNetCore.Server.Kestrel.Https;
+    using Microsoft.Extensions.Logging;
+    using Moq;
+    using Xunit;
+
+    public class BasicsStationNetworkServerTests : IAsyncLifetime
+    {
+        private const string CertificatePassword = "password";
+        private string? passwordProtectedPfx;
+        private string? notProtectedPfx;
+
+        [Theory]
+        [InlineData(true, ClientCertificateMode.NoCertificate)]
+        [InlineData(true, ClientCertificateMode.AllowCertificate)]
+        [InlineData(true, ClientCertificateMode.RequireCertificate)]
+        [InlineData(false, ClientCertificateMode.NoCertificate)]
+        [InlineData(false, ClientCertificateMode.AllowCertificate)]
+        [InlineData(false, ClientCertificateMode.RequireCertificate)]
+        public void GivenCertificateConfiguration_ConfigureHttpsSettings_Succeeds(bool serverPfxPasswordProtected, ClientCertificateMode clientCertificateMode)
```

Comment:
Will there be integration tests/E2E tests for this scenario as well?

Reply:
There will be some E2E in the end